src/otx/core/data/entity/visual_prompting.py

```diff
@@ -137,38 +133,39 @@ class ZeroShotVisualPromptingDataEntity(OTXDataEntity):
     """Data entity for zero-shot visual prompting task.
 
     Attributes:
-        masks (tv_tensors.Mask): The masks of the instances.
-        labels (LongTensor): The labels of the instances.
-        polygons (list[Polygon]): The polygons of the instances.
-        prompts (list[tv_tensors.TVTensor]): The prompts of the instances.
+        masks (tvMask): The masks of the instances.
+        labels (dict[Literal["prompts", "polygons", "masks"], list[LongTensor]]): The labels of the instances
+            for each prompt.
+        polygons (list[dmPolygon]): The polygons of the instances.
+        prompts (list[tvBoundingBoxes | Points]): The prompts of the instances.
     """
 
     @property
     def task(self) -> OTXTaskType:
         """OTX Task type definition."""
         return OTXTaskType.ZERO_SHOT_VISUAL_PROMPTING
 
-    masks: tv_tensors.Mask
-    labels: list[LongTensor]
-    polygons: list[Polygon]
-    prompts: list[tv_tensors.TVTensor]
+    masks: tvMask
+    labels: dict[Literal["prompts", "polygons", "masks"], LongTensor]
+    polygons: list[dmPolygon]
+    prompts: list[tvBoundingBoxes | Points]
 
 
 @dataclass
 class ZeroShotVisualPromptingBatchDataEntity(OTXBatchDataEntity[ZeroShotVisualPromptingDataEntity]):
     """Data entity for zero-shot visual prompting task.
 
     Attributes:
-        masks (list[tv_tensors.Mask]): List of masks.
-        labels (list[LongTensor]): List of labels.
-        polygons (list[list[Polygon]]): List of polygons.
-        prompts (list[list[tv_tensors.TVTensor]]): List of prompts.
+        masks (list[tvMask]): List of masks.
+        labels (list[dict[Literal["prompts", "polygons", "masks"], LongTensor]]): List of labels.
+        polygons (list[list[dmPolygon]]): List of polygons.
+        prompts (list[list[tvBoundingBoxes | Points]]): List of prompts.
     """
 
-    masks: list[tv_tensors.Mask]
-    labels: list[LongTensor]
-    polygons: list[list[Polygon]]
-    prompts: list[list[tv_tensors.TVTensor]]
+    masks: list[tvMask]
+    labels: list[dict[Literal["prompts", "polygons", "masks"], LongTensor]]
```

Comment:
Isn't it time to introduce a dataclass instead of dict?

Reply:
Updated ab08d68